packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -421,126 +423,103 @@ export const makeInnerVault = (
-        client: AmountMath.add(clientAllocation, proposal.want.RUN),
-      };
-    } else if (proposal.give.RUN) {
-      // We don't allow debt to be negative, so we'll refund overpayments
-      // TODO this is the same as in `transferRun`
-      const currentDebt = getCurrentDebt();
-      const acceptedRun = AmountMath.isGTE(proposal.give.RUN, currentDebt)
-        ? currentDebt
-        : proposal.give.RUN;
-
-      return {
-        vault: acceptedRun,
-        client: AmountMath.subtract(clientAllocation, acceptedRun),
-      };
-    } else {
-      return {
-        vault: AmountMath.makeEmpty(debtBrand),
-        client: clientAllocation,
-      };
-    }
-  };
+  const applyDelta = (base, gain, loss) =>
+    AmountMath.subtract(AmountMath.add(base, gain), loss);
 
-  const transferRun = seat => {
-    const { vaultSeat } = state;
-    const proposal = seat.getProposal();
-    if (proposal.want.RUN) {
-      seat.incrementBy(
-        vaultSeat.decrementBy(harden({ RUN: proposal.want.RUN })),
-      );
-    } else if (proposal.give.RUN) {
-      // We don't allow debt to be negative, so we'll refund overpayments
-      const currentDebt = getCurrentDebt();
-      const acceptedRun = AmountMath.min(proposal.give.RUN, currentDebt);
-      vaultSeat.incrementBy(seat.decrementBy(harden({ RUN: acceptedRun })));
-    }
+  /**
+   * Calculate the fee, the amount to mint and the resulting debt.
+   * The give and the want together reflect a delta, where typpically
+   * one is zero. In that case, the `fee` will also be zero, so the
+   * simple math works
+   *
+   * @param {Amount} currentDebt
+   * @param {Amount} giveAmount
+   * @param {Amount} wantAmount
+   */
+  const loanFee = (currentDebt, giveAmount, wantAmount) => {
+    const fee = ceilMultiplyBy(wantAmount, manager.getLoanFee());
+    const toMint = AmountMath.add(wantAmount, fee);
+    const newDebt = applyDelta(currentDebt, toMint, giveAmount);
+    return { newDebt, toMint, fee };
   };
 
   /**
-   * Calculate the fee, the amount to mint and the resulting debt
+   * Check whether we can proceed with an `adjustBalances`.
    *
-   * @param {ProposalRecord} proposal
-   * @param {{vault: Amount, client: Amount}} debtAfter
+   * @param {Amount} newCollateralPre
+   * @param {Amount} maxDebtPre
+   * @param {Amount} newCollateral
+   * @param {Amount} newDebt
+   * @returns {boolean}
    */
-  const loanFee = (proposal, debtAfter) => {
-    let newDebt;
-    const currentDebt = getCurrentDebt();
-    let toMint = AmountMath.makeEmpty(debtBrand);
-    let fee = AmountMath.makeEmpty(debtBrand);
-    if (proposal.want.RUN) {
-      fee = ceilMultiplyBy(proposal.want.RUN, manager.getLoanFee());
-      toMint = AmountMath.add(proposal.want.RUN, fee);
-      newDebt = AmountMath.add(currentDebt, toMint);
-    } else if (proposal.give.RUN) {
-      newDebt = AmountMath.subtract(currentDebt, debtAfter.vault);
-    } else {
-      newDebt = currentDebt;
+  const checkRestart = (
```

Comment:
"check" doesn't imply what should happen and "restart" is not clear what's being restarted.
```suggestion
  const shouldRestartAdjustBalances = (
```

Reply:
Check doesn't care what should happen. It's just saying your info is stale. It's up to the caller what should happen. That's why I pulled that out of the name of the function (as well as to prevent spurious line wraps :D).

Reply:
> Check doesn't care what should happen.

A callee doesn't necessarily care what will happen, but since this includes "Restart" that makes it concerned with how this value will be used. The end of that function says,
```
    // The collateral went up. Restart if the debt *also* went up because
    // the price quote might not apply at the higher numbers.
```

So the function body is prescribing what should happen. My request is that you make that clear in the function name. 

> It's just saying your info is stale. It's up to the caller what should happen.

That would be fine too. Then I'd request the name be something like "isNewDebtGreater`.